Escaped EOF

tabatkins opened the pull request:
Tests <https://github.com/w3c/csswg-drafts/issues/3182> and <https://github.com/w3c/csswg-drafts/issues/1821>

tabatkins committed:
Escaped EOF

Tests <https://github.com/w3c/csswg-drafts/issues/3182> and <https://github.com/w3c/csswg-drafts/issues/1821>

domenic commented on css/css-syntax/escaped-eof.html:
```diff
@@ -0,0 +1,53 @@
+}
+
+test(()=>{
+    assert_true(threw(()=>{document.querySelector("#123");}), "numeric hash token is invalid in a selector");
```
Use assert_throws, so you can test the exception type too?

tabatkins replied:
Was using it originally, but didn't like the asymmetry in my second condition. That said... (see next response).

domenic commented on css/css-syntax/escaped-eof.html:
```diff
@@ -0,0 +1,53 @@
+
+test(()=>{
+    assert_true(threw(()=>{document.querySelector("#123");}), "numeric hash token is invalid in a selector");
+    assert_false(threw(()=>{document.querySelector("#foo\\");}), "escaped-EOF in a hash token is valid in a selector");
```
If you just perform the querySelector, any exceptions will cause automatic test failures, with better failure messages than "got true, expected false".

tabatkins committed:
Use assert_throws, etc

Per @domenic's advice, go ahead and use assert_throws on the throwing call, and then just make the should-succeed call directly, as a throw will cause the test to fail anyway.
